protocols/gossipsub/src/handler.rs

```diff
@@ -181,6 +181,13 @@ impl Handler {
             <Self as ConnectionHandler>::InboundOpenInfo,
         >,
     ) {
+        if self.inbound_substreams_created == MAX_SUBSTREAM_CREATION {
+            // Too many inbound substreams have been created, end the connection.
+            self.keep_alive = KeepAlive::No;
+            log::info!("Gossipsub error: The maximum number of inbound substreams created has been exceeded.");
+            return;
+        }
```

Comment:
Say we continuously fail to establish inbound streams. This function is only called upon successfully establishing an inbound stream. Thus, we would never set `keep_alive` to `No`, correct?

I think this check (and the one below) needs to be in `on_connection_event`. For `on_fully_negotiated_inbound` and `listen_error` we check the `inbound_substream_created` variable and the equivalent for the other.

Thoughts?

Reply:
Functionally speaking, it only makes a little difference if you look at the code.
However, semantically, these checks are better be placed in `on_connection even` indeed.
Moved.

Also, I did not manage a place where the counter is _de_ cremented.